scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -475,6 +483,48 @@ def _check(self):
                         'in lowercase: ' +
                         ', '.join(repr(x) for x in spec.enum))
 
+        # Validate the contents of compatible properties.
+        self._checked_compatibles: Set[str] = set()
+        for node in self.nodes:
+            if 'compatible' not in node.props:
+                continue
+
+            compatibles = node.props['compatible'].val
+
+            # _check() runs after _init_compat2binding() has called
+            # _dt_compats(), which already converted every compatible
+            # property to a list of strings. So we know 'compatibles'
+            # is a list, but add an assert for future-proofing.
+            assert isinstance(compatibles, list)
+
+            for compat in compatibles:
+                # This is also just for future-proofing.
+                assert isinstance(compat, str)
+
+                self._check_compatible(node, compat)
+        del self._checked_compatibles  # We have no need for this anymore.
+
+    def _check_compatible(self, node, compat):
+        if compat in self._checked_compatibles:
+            return
+
+        # The regular expression comes from dt-schema.
+        compat_re = r'^[a-zA-Z][a-zA-Z0-9,+\-._]+$'
+        if not re.match(compat_re, compat):
+            _err(f"node '{node.path}' compatible '{compat}' "
+                 'must match this regular expression: '
+                 f"'{compat_re}'")
+
```

Comment:
Skip the rest if no vendor prefix argument is given?

Reply:
I think we do skip the rest if there's no `,`, which means there is no vendor prefix.

We do want to add the compatible to `_checked_compatibles` so that we avoid repeating the check.

Reply:
I think I need to rephrase that a bit😄

I mean, if we don't pass the `--vendor-prefixes` to this script, perhaps this check should be skipped altogether? If I want to run this as part of a downstream CI run now, there's no way to pass this check if I have out-of-tree vendor prefixes, aside from copying the entire Zephyr prefix file

Reply:
D'oh, that was my intent but I did it wrong. Should be fixed now; thanks for catching that.